orderbook/src/orderbook.rs

```diff
@@ -82,9 +82,21 @@ impl Orderbook {
             Some(order) => order,
             None => return Ok(AddOrderResult::InvalidSignature),
         };
-        self.balance_fetcher
-            .register(order.order_meta_data.owner, order.order_creation.sell_token)
-            .await;
+        let owner = order.order_meta_data.owner;
```

Comment:
If we already have `owner` in the order metadata then I think this could be used in place #414 to detect a `WrongOwner` error on order creation (i.e. instead of introducing Optional from address). I have already started a PR for this #449 but we might be able to make use of this instead.